settings: fix RegisterPublicNonNegativeDurationSettingWithMaximum

otan opened the pull request:
`RegisterPublicNonNegativeDurationSettingWithMaximum` should include the
maximum. Seems broken by `ac3c72339b82d9f85fd8112977a542b87fb5e712`.

Resolves https://github.com/cockroachdb/cockroach/issues/48337.

Release note (bug fix): Re-allow
`diagnostics.forced_sql_stat_reset.interval`,
`diagnostics.sql_stat_reset.interval` and `external.graphite.interval`
to set to their maximum values (24hr, 24hr and 15min respectively). This
previously only excluded these values to be allowed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48758)
<!-- Reviewable:end -->

rohany reviewed (approved):
F I'm not the timelord it seems

otan commented:
bors r=rohany

otan committed:
settings: fix RegisterPublicNonNegativeDurationSettingWithMaximum

`RegisterPublicNonNegativeDurationSettingWithMaximum` should include the
maximum. Seems broken by `ac3c72339b82d9f85fd8112977a542b87fb5e712`.

Release note (bug fix): Re-allow
`diagnostics.forced_sql_stat_reset.interval`,
`diagnostics.sql_stat_reset.interval` and `external.graphite.interval`
to set to their maximum values (24hr, 24hr and 15min respectively). This
previously only excluded these values to be allowed.

otan commented:
bors r=rohany
